core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/OperationAuthoriser.java

```diff
@@ -111,6 +112,10 @@ public void setAuthsFromStrings(final Map<String, Set<String>> auths) throws Cla
 
     protected void authorise(final Operation operation, final User user) {
         if (null != operation) {
+            if (operation instanceof OperationChain) {
```

Comment:
Please could you add some tests to check nested operations are handled correctly.